can/interfaces/udp_multicast/bus.py

```diff
@@ -320,21 +340,37 @@ def recv(
                 self.max_buffer, self.received_ancillary_buffer_size
             )
 
-            # fetch timestamp; this is configured in in _create_socket()
-            assert len(ancillary_data) == 1, "only requested a single extra field"
-            cmsg_level, cmsg_type, cmsg_data = ancillary_data[0]
-            assert (
-                cmsg_level == socket.SOL_SOCKET and cmsg_type == SO_TIMESTAMPNS
-            ), "received control message type that was not requested"
-            # see https://man7.org/linux/man-pages/man3/timespec.3.html -> struct timespec for details
-            seconds, nanoseconds = struct.unpack(
-                self.received_timestamp_struct, cmsg_data
-            )
-            if nanoseconds >= 1e9:
-                raise can.CanError(
-                    f"Timestamp nanoseconds field was out of range: {nanoseconds} not less than 1e9"
+            # fetch timestamp; this is configured in _create_socket()
+            if self.timestamp_nanosecond:
+                # Very similar to timestamp handling in can/interfaces/socketcan/socketcan.py -> capture_message()
+                assert len(ancillary_data) == 1, "only requested a single extra field"
```

Comment:
assert statements are ignored if python is run with the [-o option](https://docs.python.org/3/using/cmdline.html#cmdoption-O)

Reply:
Okay, you are right. But that's how it is done in *socketcan* (and in a single line in `can/io/blf.py`) too. Shall I change those too?

Reply:
No, at least not in this PR. But generally production code should not contain assert statements.

Reply:
Jup, you're right. Maybe a linter could check that.